command/agent/rpc.go

```diff
@@ -616,6 +626,25 @@ func (i *AgentRPC) handleSerfQuery(client *rpcClient, seq uint64) error {
 	return client.Send(&resp, nil)
 }
 
+func (i *AgentRPC) handleSerfPing(client *rpcClient, seq uint64) error {
+	var req serfPingRequest
+	if err := client.dec.Decode(&req); err != nil {
+		return fmt.Errorf("decode failed: %v", err)
+	}
+
+	i.logger.Printf("[DEBUG] agent.rpc: handling serf ping request")
+
+	// Start the serf query
+	resp, err := i.agent.SerfPing(req.Name)
```

Comment:
Excellent point.  If the error gets returned here, it does not get propagated to the client.  An error such as the node not being in the member list would not be propagated.

I think I'd prefer something similar to the following at line 645:
```
if replyErr := client.Send(&header, &resp); replyErr != nil {
    return replyErr
else
    return err
fi
```

Would you be comfortable with this change?